src/python/pants/backend/python/tasks/resolve_requirements_task_base.py

```diff
@@ -71,13 +72,18 @@ def resolve_requirements(self, interpreter, req_libs):
       else:
         maybe_platforms = None
 
+      # If sdist_builds are allowed, we don't want the translated bdist to be cached locally
```

Comment:
Hm... I don't think comment lines up with what the next line is doing.

But moreover, not caching things at all when they've been translated (as the comment implies) would be very, very slow. So while I agree with the TODO, I don't think that disabling cache_distributions in this case makes sense.